execution-engine/contracts/hdac-system/pop/src/contract_delegations.rs

```diff
@@ -102,6 +103,49 @@ impl ContractDelegations {
             runtime::put_key(&name, Key::Hash([0; 32]));
         }
     }
+
+    pub fn read_user_stat() -> Result<DelegationStat> {
+        let mut delegation_stat = BTreeMap::new();
+        for (name, _) in runtime::list_named_keys() {
+            let mut split_name = name.split('_');
+            if Some("d") != split_name.next() {
+                continue;
+            }
+
+            let to_publickey = |hex_str: &str| -> Result<PublicKey> {
+                if hex_str.len() != 64 {
+                    return Err(Error::DelegationsKeyDeserializationFailed);
+                }
+                let mut key_bytes = [0u8; 32];
+                let _bytes_written = base16::decode_slice(hex_str, &mut key_bytes)
+                    .map_err(|_| Error::DelegationsKeyDeserializationFailed)?;
+                debug_assert!(_bytes_written == key_bytes.len());
+                Ok(PublicKey::from(key_bytes))
+            };
+
+            let hex_key = split_name
+                .next()
+                .ok_or(Error::DelegationsKeyDeserializationFailed)?;
+            let delegator = to_publickey(hex_key)?;
+
+            let hex_key = split_name
+                .next()
+                .ok_or(Error::DelegationsKeyDeserializationFailed)?;
+            let _validator = to_publickey(hex_key)?;
```

Comment:
It seems we can just skip them.

Reply:
https://github.com/hdac-io/CasperLabs/pull/14/commits/c60a3ae20af28135a5db18a4f451e0836c2c0621

Reply:
We can skip the elements by using `nth(n)`.

```rust
let hex_key = split_name
    .next()
    .ok_or(Error::DelegationsKeyDeserializationFailed)?;
let delegator = to_publickey(hex_key)?;

let balance = split_name
    .nth(1)
    .and_then(|b| U512::from_dec_str(b).ok())
    .ok_or(Error::DelegationsDeserializationFailed)?;
```

Reply:
Does it means:

```rust
let balance = split_name.nth(0) // <- == next()
```

?

Reply:
Isn't `nth(0)` a validator public key in this context?

Reply:
Then, `split_name.nth(x)` is xth array of `split_name`, right?

Reply:
Nope. `nth()` is iterator.
https://users.rust-lang.org/t/split-and-nth/29795

Reply:
I don't understand what you are saying.
What I meant was we can reduce dummy `next()` call by calling `nth(1)`.

The thread you mention is about calling `nth(n)` method redundantly.

Reply:
Confirmed that resolved in 27de20c.

Reply:
I meant, how I could process nth item of array.
According to doc, if `split_name` is `['v', 'xxxx', 'yyyy']`, `split_name.nth(0), split_name.nth(0), split_name.nth(0)` if I want to get all members

Reply:
Ah, now I see. :+1: